src/main/java/duke/parser/Command.java

```diff
@@ -0,0 +1,136 @@
+package duke.parser;
+
+import duke.task.TaskList;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.ToDoTask;
+import duke.storage.Storage;
+import duke.ui.UiText;
+
+import java.util.Scanner;
+
+import java.util.List;
+
+import java.util.stream.Collectors;
+
+public class Command {
+    Scanner terms;
+
+    Command(Scanner sc) {
+        this.terms = sc;
+    }
+
+    /**
+     * Add new user task into task list
+     *
+     * @param first   = task type specified by user
+     * @param terms   = subsequent terms specified by to define the task
+     * @param tasks   = task list
+     * @param ui      = ui for responding to user after adding the new task
+     * @param storage = save to local file after every changes to task list
+     * @return task successfully added?
+     */
+    static boolean addTask(String first, Scanner terms, TaskList tasks, UiText ui, Storage storage) {
+        Task t = null;
+        try {
+            switch (first) {
+                case "todo":
+                    t = new ToDoTask(terms.nextLine().trim());
+                    break;
+                case "deadline":
+                    t = new DeadlineTask(terms.useDelimiter("/").next().trim(), terms.useDelimiter(" ").next().substring(1),

```

Comment:
Extracting out this repeated string formatting as a method will help you avoid regression errors in future as the change in future will require only updating that method.